sdk/storage/azblob/internal/exported/user_delegation_credential.go

```diff
@@ -0,0 +1,49 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package exported
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/generated"
+)
+
+// NewUserDelegationCredential creates a new UserDelegationCredential using a Storage account's Name and a user delegation Key from it
+func NewUserDelegationCredential(accountName string, udk generated.UserDelegationKey) *UserDelegationCredential {
+	return &UserDelegationCredential{
+		Name: accountName,
+		Key:  udk,
+	}
+}
+
+// UserDelegationKey contains UserDelegationKey.
+type UserDelegationKey = generated.UserDelegationKey
+
+// UserDelegationCredential contains an account's name and its user delegation key.
+type UserDelegationCredential struct {
+	Name string
+	Key  UserDelegationKey
+}
+
+// AccountName returns the Storage account's Name
+func (f *UserDelegationCredential) AccountName() string {
+	return f.Name
+}
+
+// ComputeHMAC
+func (f *UserDelegationCredential) ComputeHMACSHA256(message string) (string, error) {
```

Comment:
This method, and `GetUDKParams` below shouldn't be on the `UserDelegationCredential` as they become part of the public surface area.  Can you make them funcs instead?  This way, only the SDK can call them.